### ISSUE TITLE
Page value is only available after content dimensions are established

### ISSUE THREAD

**Issue by ETalha** [serenader2014/flutter_carousel_slider#210]
this error is being thrown while running my flutter code of this amazing package
**AssertionError ('package:flutter/src/widgets/page_view.dart': Failed assertion: line 373 pos 7: 'pixels == null || (minScrollExtent != null && maxScrollExtent != null)': Page value is only available after content dimensions are established.)**

my code is 
`class Slideshow extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: CarouselSlider(
        options: CarouselOptions(
          autoPlay: true,
          enlargeCenterPage: true,
          autoPlayCurve: Curves.easeInOut,
          aspectRatio: 16 / 9,
          viewportFraction: .75,
        ),
        items: [
          "https://firebasestorage.googleapis.com/v0/b/san-67ce2.appspot.com/o/clothes.jpeg?alt=media&token=d3be6bcb-4a40-4c78-a577-0b4f6a386536",
          "https://firebasestorage.googleapis.com/v0/b/san-67ce2.appspot.com/o/clothes.jpeg?alt=media&token=d3be6bcb-4a40-4c78-a577-0b4f6a386536",
          "https://firebasestorage.googleapis.com/v0/b/san-67ce2.appspot.com/o/clothes.jpeg?alt=media&token=d3be6bcb-4a40-4c78-a577-0b4f6a386536",
          "https://firebasestorage.googleapis.com/v0/b/san-67ce2.appspot.com/o/clothes.jpeg?alt=media&token=d3be6bcb-4a40-4c78-a577-0b4f6a386536",
          "https://firebasestorage.googleapis.com/v0/b/san-67ce2.appspot.com/o/clothes.jpeg?alt=media&token=d3be6bcb-4a40-4c78-a577-0b4f6a386536",
        ].map((i) {
          return Builder(
            builder: (BuildContext context) {
              return Padding(
                padding: const EdgeInsets.all(2.0),
                child: Container(
                    width: MediaQuery.of(context).size.width,
                    margin: EdgeInsets.symmetric(horizontal: 5.0),
                    decoration: BoxDecoration(
                        color: Colors.white,
                        border: Border.all(color: Colors.black)),
                    child: Image.network(
                      i,
                      fit: BoxFit.fill,
                      alignment: Alignment.center,
                    )),
              );
            },
          );
        }).toList(),
      ),
    );
  }
}`

**Comment by digitalistiqamah** [comment]
I got same problem. For temporary solution, just use previous version (without ^): 
=> carousel_slider: 2.2.1

**Comment by 2shrestha22** [comment]
Yes same problem with latest version. Problem is triggered when using `enlargeCenterPage: true`.

**Comment by DengpanWang** [comment]
the problem will disappear if running release mode, even setting `enlargeCenterPage = true`. However, in debug mode, `enlargeCenterPage = true` will try to access the `PageController.page` that cannot be accessed before the first build of items. Although the author uses `try ... catch ..` to avoid failing to access `PageController.page`, it seems not working on debug mode.

**Comment by diasmashikov** [comment]
I have the same problem with the latest version of Flutter 2.10. Are there any solutions for the debug mode?

**Comment by tmushayahama** [comment]
Any solution to this? I am running carousel_slider 4.0.0

**Comment by chantellewis** [comment]
Hi I'm having the same problem, is there any solutions to this?

**Comment by braglath** [comment]
if you dont worry about debug mode then we can use 
`enlargeCenterPage: kDebugMode ? false : true,`

any solution for carousel_slider 4.0.0 debug mode

**Comment by haizadvnet** [comment]
Version : carousel_slider 4.0.0

Maybe we need to add some checking for dimensions in carousel_slider.dart such as
```
carouselState!.pageController?.position?.haveDimensions == true
```

I already tried code below, but it doesn't seem smooth.
```
if (carouselState!.pageController?.position?.haveDimensions == true && widget.options.enlargeCenterPage != null &&
                widget.options.enlargeCenterPage == true) {
```
